src/main/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -492,21 +494,30 @@ void enableFileSource(final SamReader reader, final boolean enabled) {
         return BAMFileSpan.merge(spanArray).toCoordinateArray();
     }
 
-    private class CRAMIntervalIterator extends BAMQueryMultipleIntervalsIteratorFilter
+    private abstract class CRAMIntervalIteratorBase extends BAMQueryMultipleIntervalsIteratorFilter
             implements CloseableIterator<SAMRecord> {
 
         // the granularity of this iterator is the container, so the records returned
         // by it must still be filtered to find those matching the filter criteria
         private CRAMIterator unfilteredIterator;
         SAMRecord nextRec = null;
 
-        public CRAMIntervalIterator(final QueryInterval[] queries, final boolean contained) {
+        public CRAMIntervalIteratorBase(final QueryInterval[] queries, final boolean contained) {
             this(queries, contained, coordinatesFromQueryIntervals(getIndex(), queries));
         }
 
-        public CRAMIntervalIterator(final QueryInterval[] queries, final boolean contained, final long[] coordinates) {
+        public CRAMIntervalIteratorBase(final QueryInterval[] queries, final boolean contained, final long[] coordinates) {
             super(queries, contained);
+        }
 
+        /**
+         * Subclasses must call this method in their constructors AFTER construction of this class is complete.
```

Comment:
could you not have a "abstract callInitialize()" method in the base class and then call that method from the base constructor? that would force an subclass to implement this method and read this comment...otherwise I'm not sure it will be noticed....

Reply:
I agree that this is error prone but at least it's a private class, so it's likely a future subclasser will read this comment before writing code.

I don't see why this class still has `CRAMIntervalIteratorBase(final QueryInterval[] queries, final boolean contained, final long[] coordinates)` as `coordinates` is ignored. Removing that argument will also be a way to tell callers that they must call this method, which you could rename to something like `setCoordinates()`.

Reply:
Yes, that constructor appears to be redundant. Removed. Not sure I understand how that helps with the requirement to call initialize though.